### PR TITLE
MG-1004: Remap RNA expression sex display values to singluar format

### PR DESCRIPTION
**Pull request by JessterB**
# Problem

The Sex column in the Disease Correlation CT uses singular sex values (Female, Male), but the new Sex column in the RNA DE CT is displaying plural values (Females, Males). The individual RNA results page also displays the plural values.

# Solution

Update the rna_de_aggregate and rna_de_individual custom transforms to map the plural versions to the singular form, while allowing unmapped sex values (such as those from any results files that already contain the singular form) to pass through unaltered.

Note: Depending on merge order, either this PR or the [rna_de GX PR](https://github.com/Sage-Bionetworks/agora-data-tools/pull/369) will need to update the GX allowed values for this field.

### CODE REVIEW

File: src/agoradatatools/etl/transform/transform_utils/rna_de_individual_utils.py [hunk 1]
```diff
@@ -168,6 +169,9 @@ def preprocess_data_file(
     data_file["tissue"] = data_file["tissue"].str.replace(
         "Right Cerebral Hemisphere", "Hemibrain", regex=False
     )
+    # Map plural source sex labels to their singular display form
+    SEX_DISPLAY_MAP = {"Females": "Female", "Males": "Male"}
+    data_file["sex"] = data_file["sex"].replace(SEX_DISPLAY_MAP)
```
Comment (jaclynbeck-sage) [inline src/agoradatatools/etl/transform/transform_utils/rna_de_individual_utils.py]: Nit: I don't think you need to assign to a variable since it's never used again. You can just use it directly:
```
data_file["sex"] = data_file["sex"].replace({"Females": "Female", "Males": "Male"})
```

Comment (jaclynbeck-sage) [reply src/agoradatatools/etl/transform/transform_utils/rna_de_individual_utils.py]: Actually, to make things consistent, I would suggest making a one-line helper function in model_ad_transform_utils.py:

```
def remap_sex_labels(column: pd.Series) -> pd.Series:
    return column.replace({"Females": "Female", "Males": "Male"})
```
and calling it both here and in the other file that needs this (`data_file["sex"] = remap_sex_labels(data_file["sex"])`)

Then it can have its own unit tests and we don't need to expand the test suites for the transforms themselves.

File: src/agoradatatools/etl/transform/rna_de_aggregate.py [hunk 2]
```diff
@@ -424,6 +429,9 @@ def _process_single_data_file(
     # Filter out combined-cohort rows; keep single-sex rows only
     data_file = data_file[data_file["sex"] != "Females & Males"]
 
+    # Map plural source sex labels to their singular display form
+    data_file = data_file.assign(sex=data_file["sex"].replace(SEX_DISPLAY_MAP))
```
Comment (jaclynbeck-sage) [inline src/agoradatatools/etl/transform/rna_de_aggregate.py]: See note above about creating a helper function instead.

File: tests/transform/test_rna_de_aggregate.py [hunk 3]
```diff
@@ -1029,11 +1030,107 @@ def test_process_single_data_file_filters_combined_sex_cohort(self) -> None:
...
+            label_map_dict=label_map_dict,
+            model_group_dict={},
+            biodomain_dict={},
+            model_type_dict={},
+            file_index=0,
+            total_files=1,
+        )
+
+        assert {entry["sex"] for entry in result} == {"Male", "Female"}
+        assert not any(entry["sex"] in {"Females", "Males"} for entry in result)
```
Comment (jaclynbeck-sage) [inline tests/transform/test_rna_de_aggregate.py]: I think this last assert statement is redundant: if the first assert passes, then there aren't any other values besides "Male" and "Female".

File: tests/transform/test_rna_de_individual.py [hunk 4]
```diff
@@ -43,10 +43,73 @@
...
+            file_name="sex_mapping.csv",
+            data_file=data_file,
+            file_index=0,
+            total_files=1,
+            data_file_required_columns=DATA_FILE_REQUIRED_COLUMNS,
+            data_file_column_rules=DATA_FILE_COLUMN_RULES,
+        )
+
+        assert set(result["sex"]) == {"Female", "Male"}
+        assert not result["sex"].isin(["Females", "Males"]).any()
```
Comment (jaclynbeck-sage) [inline tests/transform/test_rna_de_individual.py]: Redundant assert

**Review by jaclynbeck-sage — CHANGES_REQUESTED** [review]
This looks great! I have a few nitpicks but it's almost there.

File: tests/transform/test_rna_de_aggregate.py [hunk 5]
```diff
@@ -1029,11 +1030,107 @@ def test_process_single_data_file_filters_combined_sex_cohort(self) -> None:
...
+
+    def test_process_single_data_file_singular_sex_values_pass_through(self) -> None:
+        """Singular source sex labels are passed through without alteration")."""
```
Comment (jaclynbeck-sage) [inline tests/transform/test_rna_de_aggregate.py]: Remove extra parenthesis at the end

File: src/agoradatatools/etl/transform/rna_de_aggregate.py [hunk 6]
```diff
@@ -8,6 +8,7 @@
 The transformation:
 - Filters to mouse genes only (ENSMUSG*), excluding human genes (ENSG*)
 - Filters out combined-cohort rows where sex is "Females & Males", keeping single-sex rows only
+- Maps plural sex labels to singular display values ("Females" -> "Female", "Males" -> "Male")
```
Comment (linglp) [inline src/agoradatatools/etl/transform/rna_de_aggregate.py]: I saw that in this PR #369 [here](https://github.com/Sage-Bionetworks/agora-data-tools/pull/369/changes#diff-c2022b5a8d695f73e8ad9fd36cc1303b596f6870eb3a823dfdf547ad048864ad), that we have: 
```
    {
      "expectation_type": "expect_column_values_to_be_in_set",
      "kwargs": {
        "column": "sex",
        "value_set": [
          "Females",
          "Males"
        ]
      },
      "meta": {}
    },
```
see [here](https://github.com/Sage-Bionetworks/agora-data-tools/blob/df5e2a54db6ee241436ee6cac9dcf267d5ca0f8d/src/agoradatatools/great_expectations/gx/expectations/rna_de_aggregate.json#L154-L161) in `src/agoradatatools/great_expectations/gx/expectations/rna_de_aggregate.json`

Does the change here conflict with the expectation rule?

Comment (JessterB) [reply src/agoradatatools/etl/transform/rna_de_aggregate.py]: Yes, I wasn't sure which PR would get merged first  so I have notes in both PR descriptions about updating the expectations once this change is merged.

File: tests/transform/test_rna_de_aggregate.py (hunk 5 above)
Comment (JessterB) [reply tests/transform/test_rna_de_aggregate.py]: The test case has been removed

File: src/agoradatatools/etl/transform/transform_utils/rna_de_individual_utils.py (hunk 1 above)
Comment (JessterB) [reply src/agoradatatools/etl/transform/transform_utils/rna_de_individual_utils.py]: Done!

File: src/agoradatatools/etl/transform/rna_de_aggregate.py (hunk 2 above)
Comment (JessterB) [reply src/agoradatatools/etl/transform/rna_de_aggregate.py]: Done!

File: tests/transform/test_rna_de_individual.py (hunk 4 above)
Comment (JessterB) [reply tests/transform/test_rna_de_individual.py]: This test no longer exists

File: tests/transform/transform_utils/test_rna_de_individual_utils.py [hunk 7]
```diff
@@ -191,6 +191,48 @@ def test_expression_already_float_unchanged(self) -> None:
...
+                "sex": sex_values,
+                "tissue": ["Hemibrain"] * n,
+                "individualid": [f"ID{i}" for i in range(n)],
+            }
+        )
+
+    @classmethod
+    def _preprocess(cls, sex_values: list[str]) -> pd.Series:
+        df = cls._make_df(sex_values)
+        result = preprocess_data_file("test.csv", df, 0, 1, cls._REQUIRED_COLUMNS, {})
```
Comment (jaclynbeck-sage) [inline tests/transform/transform_utils/test_rna_de_individual_utils.py]: Nit: since we are not testing whether it catches missing columns in this test, you can just pass an empty list instead of `cls._REQUIRED_COLUMNS`

File: tests/transform/transform_utils/test_rna_de_individual_utils.py [hunk 8]
```diff
@@ -191,6 +191,48 @@ def test_expression_already_float_unchanged(self) -> None:
...
+                "genotype": ["Tg"] * n,
+                "age": ["4 months"] * n,
+                "sex": sex_values,
+                "tissue": ["Hemibrain"] * n,
+                "individualid": [f"ID{i}" for i in range(n)],
+            }
+        )
+
+    @classmethod
+    def _preprocess(cls, sex_values: list[str]) -> pd.Series:
```
Comment (jaclynbeck-sage) [inline tests/transform/transform_utils/test_rna_de_individual_utils.py]: Nit: I'm confused why this is a separate function instead of incorporated into test_maps_sex_values, since it's only used there.

File: tests/transform/transform_utils/test_rna_de_individual_utils.py [hunk 9]
```diff
@@ -191,6 +191,48 @@ def test_expression_already_float_unchanged(self) -> None:
...
+        "model",
+        "genotype",
+        "age",
+        "sex",
+        "tissue",
+        "individualid",
+    ]
+
+    @staticmethod
+    def _make_df(sex_values: list[str]) -> pd.DataFrame:
```
Comment (jaclynbeck-sage) [inline tests/transform/transform_utils/test_rna_de_individual_utils.py]: Nit: I'm confused why this is a separate function instead of incorporated into test_maps_sex_values, since it's only used there.

File: tests/transform/transform_utils/test_rna_de_individual_utils.py [hunk 10]
```diff
@@ -191,6 +191,48 @@ def test_expression_already_float_unchanged(self) -> None:
...
+        )
+
+    @classmethod
+    def _preprocess(cls, sex_values: list[str]) -> pd.Series:
+        df = cls._make_df(sex_values)
+        result = preprocess_data_file("test.csv", df, 0, 1, cls._REQUIRED_COLUMNS, {})
+        return result["sex"].reset_index(drop=True)
+
+    def test_maps_sex_values(self) -> None:
+        """'Plurals sex values are mapped to singular display labels"""
```
Comment (jaclynbeck-sage) [inline tests/transform/transform_utils/test_rna_de_individual_utils.py]: Per Copilot, remove extra single quote and change "Plural**s**" to "Plural"

File: tests/transform/transform_utils/test_model_ad_transform_utils.py [hunk 11]
```diff
@@ -405,3 +406,38 @@ def test_validate_jax_ids_should_fail_on_invalid_formats(
...
+
+
+class TestRemapSexLabels:
+    """
+    This class tests the remap_sex_labels function to ensure that it converts plural labels to singular, and does
+    not modify labels that are already singular.
+    """
+
+    @pytest.mark.parametrize(
+        "input_ids, expected_output",
```
Comment (jaclynbeck-sage) [inline tests/transform/transform_utils/test_model_ad_transform_utils.py]: I agree with Copilot, the `input_ids` argument should be changed to something like `input_sex_values`

File: src/agoradatatools/etl/transform/transform_utils/model_ad_transform_utils.py [hunk 12]
```diff
@@ -182,3 +183,17 @@ def validate_jax_ids(jax_id: pd.Series) -> None:
...
+    Converts plural sex values ("Females" or "Males") to singular form ("Female" or "Male"). Sex values that are
+    already singular, and any other value, are not modified.
+
+    Args:
+        sex (pd.Series): A pandas Series containing sex labels that may need to be converted.
+
+    Returns:
+        pd.Series: A pandas Series containing sex labels in only the singular form.
+    """
+    return sex.replace({"Females": "Female", "Males": "Male"})
```
Comment (jaclynbeck-sage) [inline src/agoradatatools/etl/transform/transform_utils/model_ad_transform_utils.py]: Per Copilot, I would do `sex.copy().replace(...)` to avoid a setting with copy warning

File: tests/transform/transform_utils/test_model_ad_transform_utils.py (hunk 11 above)
Comment (JessterB) [reply tests/transform/transform_utils/test_model_ad_transform_utils.py]: renamed

File: src/agoradatatools/etl/transform/transform_utils/model_ad_transform_utils.py (hunk 12 above)
Comment (JessterB) [reply src/agoradatatools/etl/transform/transform_utils/model_ad_transform_utils.py]: done

File: tests/transform/transform_utils/test_rna_de_individual_utils.py (hunk 9 above)
Comment (JessterB) [reply tests/transform/transform_utils/test_rna_de_individual_utils.py]: I've collapsed the setup functions into the test case & fixed the doc string
